src/components/plan/PlanForm.svelte

```diff
@@ -130,12 +138,91 @@
       effects.updatePlan(plan, { name: $planNameField.value }, user);
     }
   }
+
+  async function downloadPlan() {
+    if (plan) {
+      if (planDownloadAbortController) {
+        planDownloadAbortController.abort();
+      }
+
+      planDownloadAbortController = new AbortController();
+
+      let qualifiedActivityDirectives: ActivityDirective[] = [];
+      planDownloadProgress = 0;
+
+      let totalProgress = 0;
+      const numOfDirectives = Object.values(activityDirectivesMap).length;
+
+      qualifiedActivityDirectives = await Promise.all(
+        Object.values(activityDirectivesMap).map(async activityDirective => {
+          if (plan) {
+            const effectiveArguments = await effects.getEffectiveActivityArguments(
+              plan?.model_id,
+              activityDirective.type,
+              activityDirective.arguments,
+              user,
+              planDownloadAbortController?.signal,
+            );
+
+            totalProgress++;
+            planDownloadProgress = (totalProgress / numOfDirectives) * 100;
+
+            return {
+              ...activityDirective,
+              arguments: effectiveArguments?.arguments ?? activityDirective.arguments,
+            };
+          }
+
+          totalProgress++;
+          planDownloadProgress = (totalProgress / numOfDirectives) * 100;
+
+          return activityDirective;
+        }),
+      );
+
+      const planDownload: PlanTransfer = getPlanForTransfer(plan, qualifiedActivityDirectives);
+
+      const a = document.createElement('a');
+      a.href = URL.createObjectURL(new Blob([JSON.stringify(planDownload, null, 2)], { type: 'application/json' }));
+      a.download = planDownload.name;
+      a.click();
+
+      planDownloadProgress = null;
+    }
+  }
+
+  function cancelPlanDownload() {
+    planDownloadAbortController?.abort();
+    planDownloadAbortController = null;
+  }
+
+  function onPlanDownload() {
+    if (planDownloadProgress === null) {
+      downloadPlan();
+    } else {
+      cancelPlanDownload();
+    }
+  }
 </script>
 
 <div class="plan-form">
   {#if plan}
     <fieldset>
       <Collapse title="Details">
+        <svelte:fragment slot="right">
+          <button
+            class="st-button icon download"
+            on:click={onPlanDownload}
+            use:tooltip={{ content: planDownloadProgress === null ? 'Download Plan JSON' : 'Cancel Plan Download' }}
+          >
+            {#if planDownloadProgress !== null}
+              <ProgressRadial progress={planDownloadProgress} useBackground={false} />
+              <div class="cancel"><CloseIcon /></div>
+            {:else}
+              <DownloadIcon />
```

Comment:
Clicking on the download icon causes the Details Collapse to trigger a collapse.